cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java

```diff
@@ -111,37 +112,88 @@ private static AmazonDynamoDB defaultDynamoDbClient() {
+    private PublicationRepresentations mapEventToNvaPublication(
+        FileContentsEvent<JsonNode> eventbody) {
+        var cristinObject = parseCristinObject(eventbody);
+        var publication = cristinObject.toPublication();
+        return new PublicationRepresentations(cristinObject, publication, eventbody);
     }
 
-    private Publication persistCristinIdentifierInFileNamedWithPublicationIdentifier(Publication publication,
-                                                                                     FileContentsEvent<JsonNode> eventBody) {
-        var cristinIdentifier =
-            getCristinIdentifier(publication);
-        var fileUri = constructSuccessFileUri(eventBody, publication);
+    private void persistPartOfCristinIdentifierIfPartOfExists(
+        PublicationRepresentations publicationRepresentations) {
+        if (cristinObjectIsPartOfAnotherPublication(publicationRepresentations.getCristinObject())) {
+            persistPartOfCristinIdentifierWithPublicationId(publicationRepresentations);
+        }
+    }
+
+    private boolean cristinObjectIsPartOfAnotherPublication(CristinObject cristinObject) {
+        return nonNull(cristinObject.getBookOrReportPartMetadata()) && nonNull(
+            cristinObject.getBookOrReportPartMetadata().getPartOf());
+    }
+
+    private String getPartOfCristinIdentifier(CristinObject cristinObject) {
+        return cristinObject.getBookOrReportPartMetadata().getPartOf();
+    }
+
+    private void persistPartOfCristinIdentifierWithPublicationId(
+        PublicationRepresentations publicationRepresentations) {
+        var partOf = createPartOf(publicationRepresentations);
+        var fileUri = constructPartOfFileUri(publicationRepresentations);
+        var s3Driver = new S3Driver(s3Client, fileUri.getUri().getHost());
+        attempt(() -> s3Driver.insertFile(fileUri.toS3bucketPath(), partOf.toJsonString())).orElseThrow();
+    }
+
+    private NvaPublicationPartOfCristinPublication createPartOf(PublicationRepresentations publicationRepresentations) {
+        var publicationIdentifier = publicationRepresentations.getPublication().getIdentifier().toString();
+        var publicationIsPartOfThisCristinPublication =
+            getPartOfCristinIdentifier(publicationRepresentations.getCristinObject());
+        return
+            NvaPublicationPartOfCristinPublication.builder()
+                .withNvaPublicationIdentifier(publicationIdentifier)
+                .withPartOf(
+                    NvaPublicationPartOf.builder()
+                        .withCristinId(publicationIsPartOfThisCristinPublication)
+                        .build())
+                .build();
+    }
+
+    private void persistCristinIdentifierInFileNamedWithPublicationIdentifier(
+        PublicationRepresentations publicationRepresentations) {
+        var cristinIdentifier = publicationRepresentations.getCristinObject().getId().toString();
```

Comment:
I'd be tempted to move this to an accessor method in the PublicationRepresentations object

Reply:
Fikset